### DISCUSSION TITLE
Returned object is not assignable to defined return type

### DISCUSSION THREAD

**Discussion by zaunermax** [millsp/ts-toolbelt#179] [Wiki]
#### Describe the bug
I seem to run into a bug, where `Update` works flawlessly on static types, but breaks on generic types.

#### Reproduce the bug
```ts
enum TypeEnum {
	FOO = 'foo',
	BAR = 'bar',
}

type HasCompany = { company: { type: string } };
type UpdatedHasCompany = Object.P.Update<HasCompany, ['company', 'type'], TypeEnum>;

// this works flawlessly
const test: UpdatedHasCompany = { company: { type: TypeEnum.BAR } };

// this does not work!
const updateType = <T extends HasCompany>(
	e: T,
): Object.P.Update<T, ['company', 'type'], TypeEnum> => ({
	...e,
	company: {
		type: TypeEnum.BAR, // here i would have a conversion function
	}
});
```

#### Expected behavior
`P.Update` should work with generics. Not sure if this is a TS limitation though :thinking:

#### Additional context
toolbelt version 8.0.7 - Typescript version 4.0.3

**Comment by millsp** [comment]
Hey @zaunermax, thanks for reporting this. This is a limitation where TypeScript is not able to predict the output in the presence of a generic type. You will have to override and cast your returned value to `any`.

**Comment by zaunermax** [comment]
Ok thanks for clarifying, sorry to having bothered you :raised_hands:

**Comment by millsp** [comment]
My pleasure, no problem. I'm adding this to the wiki.

Cheers
